orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/UpdateLaunchConfigTask.groovy

```diff
@@ -39,29 +43,43 @@ class ModifyAsgLaunchConfigurationTask implements Task, DeploymentDetailsAware {
 
   @Override
   TaskResult execute(Stage stage) {
-    def operationConfig = new HashMap(stage.context)
-    operationConfig.amiName = getImage(stage)
+    def ops
+    def cloudProvider = getCloudProvider(stage)
+    if (cloudProvider == "aws") {
+      // Since AWS is the only provider that needs to do anything here, it felt like overkill to do the
```

Comment:
Hmmm ... I'm on the fence. 

Not a huge fan of having the `aws` stuff left in the generic task. Is the alternative something along the lines of `ServerGroupCreator`?